### PR TITLE
CamSripterAPI for Axis camera events generation added.

### PR DESCRIPTION
**Pull request by CamStreamer**

### CODE REVIEW

File: src/CamScripterAPI.ts [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        this.port = options?.port ?? (this.tls ? 443 : 80);
+        this.auth = options?.auth ?? '';
+
+        this.callId = 0;
+        this.sendMessages = {};
+
+        EventEmitter.call(this);
+    }
+
+    async cameraEventsConnect() {
```
Comment (survilla) [inline src/CamScripterAPI.ts:88]: I suggest to name this mechanism like `cameraEventsGenerator` (see the class name suggestion). The name of this function would be just plain `connect()`.

File: src/CamScripterAPI.ts [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    command: string;
+    data: any;
+};
+
+type AsyncMessage = {
+    resolve;
+    reject;
+};
+
+export class CamScripterAPI extends EventEmitter {
```
Comment (survilla) [inline src/CamScripterAPI.ts:62]: can we name this class like `CamScripterAPICameraEventsGenerator` ?

**Comment by survilla** [comment]
we should update the readme.md as well

File: src/CamScripterAPI.ts (hunk 2 above)
Comment (CamStreamer) [reply src/CamScripterAPI.ts]: The name is extremely long and it looks a little bit strange in the code but it is descriptive and probably right direction when we want to split CO API as well. Changed!

File: src/CamScripterAPI.ts (hunk 1 above)
Comment (CamStreamer) [reply src/CamScripterAPI.ts]: fixed

**Review by survilla — APPROVED** [review]

**Review by survilla — APPROVED** [review]

**Pull request merged**